frontend/src/containers/MetricsStep/DeploymentFrequencySettings/SingleSelection/index.tsx

```diff
@@ -29,7 +32,7 @@ export const SingleSelection = ({ options, label, value, id, onGetSteps, onUpDat
   };
 
   useEffect(() => {
-    if (onGetSteps && !!selectedOptions) {
+    if (onGetSteps && !!selectedOptions && shouldRefresh) {
```

Comment:
React Hook useEffect has a missing dependency: 'shouldRefresh'. Either include it or remove the dependency array